ambry-store/src/test/java/com.github.ambry.store/BlobStoreStatsTest.java

```diff
@@ -241,8 +315,351 @@ public void testValidDataSizeAfterDeletes() throws InterruptedException, StoreEx
+    int bucketCount = 2 * (int) (logSegmentForecastOffsetMs / BUCKET_SPAN_IN_MS);
+    BlobStoreStats blobStoreStats = setupBlobStoreStats(bucketCount, logSegmentForecastOffsetMs);
+    // proceed only when the scan is started
+    assertTrue("IndexScanner took too long to start", scanStartedLatch.await(5, TimeUnit.SECONDS));
+    verifyAndGetContainerValidSize(blobStoreStats, state.time.milliseconds());
+    // throttle count shouldn't change after the scan is completed if new requests are within the forecast coverage
+    long throttleCountBeforeRequests = mockThrottler.throttleCount.get();
+    for (long i = 0; i <= state.time.milliseconds() + TEST_TIME_INTERVAL_IN_MS; i += TEST_TIME_INTERVAL_IN_MS) {
+      verifyAndGetLogSegmentValidSize(blobStoreStats, new TimeRange(i, 0L));
+    }
+    // advance time to let the added puts to expire
+    long timeToLiveInMs = expiresAtInMs - state.time.milliseconds() < 0 ? 0 : expiresAtInMs - state.time.milliseconds();
+    state.advanceTime(timeToLiveInMs + Time.MsPerSec);
+    for (long i = 0; i <= state.time.milliseconds() + TEST_TIME_INTERVAL_IN_MS; i += TEST_TIME_INTERVAL_IN_MS) {
+      verifyAndGetLogSegmentValidSize(blobStoreStats, new TimeRange(i, 0L));
+    }
+    verifyAndGetContainerValidSize(blobStoreStats, state.time.milliseconds());
+    assertEquals("Throttle count shouldn't have changed", throttleCountBeforeRequests,
+        mockThrottler.throttleCount.get());
+    blobStoreStats.close();
+  }
+
+  /**
+   * Test to verify new entries after a scan are properly counted to keep the current {@link ScanResults} relevant
+   * before the next scan. In addition, verify that requests within the coverage are served without triggering any scan.
+   * @throws StoreException
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  @Test
+  public void testBucketingWithNewEntriesAfterScan() throws StoreException, InterruptedException, IOException {
+    assumeTrue(bucketingEnabled);
+    CountDownLatch scanStartedLatch = new CountDownLatch(1);
+    CountDownLatch queueProcessedLatch = new CountDownLatch(1);
+    MockThrottler mockThrottler = new MockThrottler(scanStartedLatch, new CountDownLatch(0));
+    throttlers.put(BlobStoreStats.IO_SCHEDULER_JOB_TYPE, mockThrottler);
+    BlobStoreStats blobStoreStats =
+        setupBlobStoreStats(2 * (int) (state.time.milliseconds() / BUCKET_SPAN_IN_MS), state.time.milliseconds());
+    // proceed only when the scan is started
+    assertTrue("IndexScanner took too long to start", scanStartedLatch.await(5, TimeUnit.SECONDS));
+    advanceTimeToNextSecond();
+    verifyAndGetLogSegmentValidSize(blobStoreStats, new TimeRange(state.time.milliseconds(), 0L));
+    verifyAndGetContainerValidSize(blobStoreStats, state.time.milliseconds());
+    int throttleCountBeforeRequests = mockThrottler.throttleCount.get();
+    // add 3 put with expiry
+    long expiresAtInMs = state.time.milliseconds() + TimeUnit.SECONDS.toMillis(30);
```

Comment:
instead of arbitrarily setting the expiry value to 30 secs, can you make it a function of bucket span time and total no of buckets. so that its more readable to understand whether the bucketing covers the span time or not.